demos/documented/topology_optimization/projection/demo_projection.py

```diff
@@ -0,0 +1,233 @@
+
+
+class Delta(UserExpression):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def eval(self, values, x):
+        if near(x[0], 2.0) and near(x[1], 0.5):
+            values[0] = 3.0 / mesh.hmax()
+        else:
+            values[0] = 0.0
+
+    def value_shape(self):
+        return ()
+
+
+delta = Delta(degree=2)
+g = delta * Constant((0.0, -1.0))
+
+u = Function(V)
+v = Function(V)
+
+F = alpha * inner(sigma(u), eps(v)) * dx - dot(g, v) * ds(2)
+
+bcs = cashocs.create_dirichlet_bcs(V, Constant((0.0, 0.0)), boundaries, 1)
+
+# -
+
+# ### Setup of the Cost Functional and Topological Derivative
+#
+# To define the topology optimization problem, we need to define the cost functional
+# of the problem, which is given by
+
+J = cashocs.IntegralFunctional(alpha * inner(sigma(u), eps(u)) * dx)
+
+# Finally, we specify the generalized topological derivative of the problem
+# with the lines
+
+# +
+kappa = (lambd + 3.0 * mu) / (lambd + mu)
+r_in = alpha_out / alpha_in
+r_out = alpha_in / alpha_out
+
+dJ_in = Constant(
+    alpha_in * (r_in - 1.0) / (kappa * r_in + 1.0) * (kappa + 1.0) / 2.0
+) * (
+    Constant(2.0) * inner(sigma(u), eps(u))
+    + Constant((r_in - 1.0) * (kappa - 2.0) / (kappa + 2 * r_in - 1.0))
+    * tr(sigma(u))
+    * tr(eps(u))
+)
+dJ_out = Constant(
+    -alpha_out * (r_out - 1.0) / (kappa * r_out + 1.0) * (kappa + 1.0) / 2.0
+) * (
+    Constant(2.0) * inner(sigma(u), eps(u))
+    + Constant((r_out - 1.0) * (kappa - 2.0) / (kappa + 2 * r_out - 1.0))
+    * tr(sigma(u))
+    * tr(eps(u))
+)
+
+# -
+
+# The update routine for the level-set function reads
+
+
+def update_level_set():
+    cashocs.interpolate_levelset_function_to_cells(psi, alpha_in, alpha_out, alpha)
+
+
+# ### Definition and Solution of the Topology Optimization Problem
+#
+# Now, we are able to define the
+# {py:class}`TopologyOptimizationProblem <cashocs.TopologyOptimizationProblem>`
+
+top = cashocs.TopologyOptimizationProblem(
+    F, bcs, J, u, v, psi, dJ_in, dJ_out, update_level_set, config=cfg
+)
+
+# Additionally, in contrast to the previous demos, we need to specify the volume
+# constraint. For that we have to call the Projection class
+# {py:class}`projection_levelset <cashocs.projection_levelset>`. We choose the
+# lower border of the volume restriction as 0.5 and the upper one as 1.25, respectively.
+
+vol_low = 0.5
+vol_up = 1.25
+vol = [vol_low, vol_up]
+top.projection = bisection.projection_levelset(
+    top.levelset_function, volume_restriction=vol
+)
+
+# Finally, we can solve the topology optimization problem using the
+# {py:meth}`solve <cashocs.TopologyOptimizationProblem.solve>` method.
+
+top.solve(algorithm="bfgs", rtol=0.0, atol=0.0, angle_tol=1.0, max_iter=100)
+
+# :::{note}
+# In the case of an equality constraint for the volume of {math}`\Omega` we
+# have to exchange `vol` by a float describing the desired volume.
+# :::
```

Comment:
When you have a code block inside a note, you have to use one more colon to separate everything nicely. 
So it would be 
```
::::{note}
:::{code-block}
Code here
:::
::::
```

Reply:
Done!